clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfigurationProperties.groovy

```diff
@@ -59,15 +60,48 @@ class DcosConfigurationProperties {
             .withCaCertFile(cluster.caCertFile).build()
   }
 
+  private static String getServiceKeyData(Account account, ClusterCredential clusterConfig) {
+    if (clusterConfig.serviceKeyData != null && clusterConfig.serviceKeyFile != null) {
+      throw new IllegalStateException("Both a serviceKeyData and serviceKeyFile were supplied for the account with name [${account.name}] and region [${clusterConfig.name}]. Only one should be configured.")
+    } else if (clusterConfig.serviceKeyData != null) {
+        return clusterConfig.serviceKeyData
+    } else if (clusterConfig.serviceKeyFile != null) {
+      try {
+        def actualFile = new File(clusterConfig.serviceKeyFile)
+
+        return actualFile.withReader('UTF-8') {
+          it.readLines().join('\n')
+        }
+      } catch (IOException e) {
+        throw new RuntimeException(e)
+      }
+    }
+
+    // We should never reach this part of the function given that we are only calling this if one of the arguments is provided.
+    throw new IllegalStateException("Either a serviceKeyData and serviceKeyFile should be supplied for the account with name [${account.name}] and region [${clusterConfig.name}]. Neither was configured.")
+  }
+
   private static DCOSAuthCredentials buildDCOSAuthCredentials(Account account, ClusterCredential clusterConfig) {
     DCOSAuthCredentials dcosAuthCredentials = null
 
-    if (clusterConfig.uid && clusterConfig.password && clusterConfig.serviceKey) {
+    if (clusterConfig.password == null) {
```

Comment:
We still have that ability, all this does is try to load the serviceKeyData from a file if password isn't available, this could I guess be made easier if I just change the boolean logic below.